### DISCUSSION TITLE
Some questions about kura

### DISCUSSION THREAD

**Discussion by RahulKatte** [eclipse-kura/kura#4858] [Q&A] [answered]
Ours is an embedded device that runs linux. We already have our business logic implemented in java and running as plain jars. We want to use 'kura' for its wifi capabilities. We were able to build the 'generic' version of kura and install it on our device, but it comes with a lot of services that we don't need. 
	1. How do we go about building kura with just the services that we want ? in our case, we just want the 'wifi' services.
	2. If cherry picking at build time is not possible, how to we go about stopping (or not starting) certain services at runtime ?
	3. Is there a way by which other processes can connect with Kura services and invoke certain operations ? For ex, can a plain java jar running in a separate process somehow ask Kura to perform wifi related tasks ?
	4. Is Wifi P2p supported ? If yes, what authentication mechanisms are supported, PBC, PIN, etc ?
	5. Is Bluetooth supported out of the box ?
	6. Is 4G/5G LTE supported out of the box ?

**Comment by salvatore-coppola** [comment, marked as answer]
Hi @RahulKatte thank you for the interest in Kura.

I'll try to answer all your questions:
1. Kura is a modular system built on the equinox OSGi framework, each service is contained in a bundle.  You can remove the bundles and the related services from the distribution editing [the pom.xml distribution file](https://github.com/eclipse/kura/blob/develop/kura/distrib/pom.xml) and the [build_equinox_distrib file](https://github.com/eclipse/kura/blob/develop/kura/distrib/src/main/ant/build_equinox_distrib.xml). Pay attention to remove the bundles and the reference in the generated `config.ini`
2. See the point 1
3. You can develop your own OSGi bundle or deployement package and deploy it in Kura following the guides in the [official kura documentation](https://eclipse.github.io/kura/docs-release-5.3). Kura also provides REST API that can be used to invoke services from outside the jvm running Kura and equinox.
4. Wi-Fi Direct is currently not supported, but contributions are welcome if you would like to implement it. 
5. Yes but the compatibility with the specific device/chipset need to be verified, please refer to the official Kura documentation.
6. Yes but the compatibility with the specific device/chipset need to be verified, please refer to the official Kura documentation.

I hope I have helped you, bye.

Salvatore
